Library/include/CSP/Systems/Quota/QuotaSystem.h

```diff
@@ -0,0 +1,101 @@
+/*
+* Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "CSP/Systems/SystemBase.h"
+#include "Quota.h"
+
+
+namespace csp::services
+{
+
+class ApiBase;
+
+}
+
+
+namespace csp::web
+{
+
+class WebClient;
+
+}
+
+namespace csp::systems
+{
+/// @ingroup GraphQL System
+/// @brief Public facing system that allows interfacing with Magnopus Connect Services' Quota Server.
+/// Offers methods for receiving Quota Queries.
+class CSP_API CSP_NO_DISPOSE QuotaSystem : public SystemBase
+{
+	/** @cond DO_NOT_DOCUMENT */
+	friend class SystemsManager;
+	/** @endcond */
+
+	/// @brief Get total current users spaces owned and the current users space limit
+	/// @param Callback FeatureProgressCallback : callback when asynchronous task finishes
+	CSP_ASYNC_RESULT void GetTotalSpaceOwnedByUser(FeatureLimitCallback Callback);
+
+	/// @brief Gets total number of user current present inside of a space
+	/// @param SpaceId csp::common::String : Id of the Space
+	/// @param Callback FeatureProgressCallback : callback when asynchronous task finishes
+	void GetConcurrentUsersInSpace(const csp::common::String& SpaceId, FeatureLimitCallback Callback);
+
+	/// @brief Get total size of all assets within a space
+	/// @param SpaceId csp::common::String : Id of the Space
+	/// @param Callback FeatureProgressCallback : callback when asynchronous task finishes
+	void GetTotalSpaceSizeinKilobytes(const csp::common::String& SpaceId, FeatureLimitCallback Callback);
+
+	/// @brief Get Array of feature progress for a user
+	/// @param FeatureName csp::common::Array<csp::common::String> : Array of feature names that will be retrieved
+	/// @param Callback FeatureProgressCallback : callback when asynchronous task finishes
+	void GetTierFeatureProgressForUser(const csp::common::Array<csp::common::String>& FeatureName, FeaturesLimitCallback Callback);
```

Comment:
Correct me if I'm wrong... f we do go down the string path, it will be extremely useful(basically necessary) to also have a way to get a list of all tier names, and a list of all feature names. This would allow us on a client level to have a concrete source of tier names/feature names when calling these endpoints.

Reply:
@MAG-AdamThorn this is a good point

Reply:
An enum solves this for us in a more elegant client friendly way, hence the request for enums

Reply:
Yes I agree that the Clients/3rd parties will need to have a way of getting tier and feature names.
I also agree that passing enum names would be a good option for solving this.